hibernate-core/src/test/java/org/hibernate/orm/test/temporal/FractionalSecondsTests.java

```diff
@@ -0,0 +1,268 @@
+	}
+
+	@Test
+	@DomainModel(annotatedClasses = TestEntity0.class)
+	@SessionFactory
+	@SkipForDialect( dialectClass = H2Dialect.class, reason = "Occasional mismatch in rounding versus our code" )
+	@SkipForDialect( dialectClass = MariaDBDialect.class, reason = "Occasional mismatch in rounding versus our code" )
+	@SkipForDialect( dialectClass = MySQLDialect.class, reason = "Occasional mismatch in rounding versus our code", matchSubTypes = true )
+	@SkipForDialect( dialectClass = OracleDialect.class, reason = "Occasional mismatch in rounding versus our code" )
+	@SkipForDialect( dialectClass = SQLServerDialect.class, reason = "Occasional mismatch in rounding versus our code" )
+	@SkipForDialect( dialectClass = PostgreSQLDialect.class, reason = "Occasional mismatch in rounding versus our code", matchSubTypes = true )
+	@SkipForDialect( dialectClass = DerbyDialect.class, reason = "Derby does not support sized timestamp" )
+	@SkipForDialect(dialectClass = SybaseDialect.class, reason = "Because... Sybase...", matchSubTypes = true)
+	void testUsage0(SessionFactoryScope scope) {
+		final Instant start = Instant.now();
+
+		scope.inTransaction( (session) -> {
+			final TestEntity0 testEntity = new TestEntity0();
+			testEntity.id = 1;
+			testEntity.theInstant = start;
+			session.persist( testEntity );
+		} );
+
+		scope.inTransaction( (session) -> {
+			final TestEntity0 testEntity = session.find( TestEntity0.class, 1 );
+			assertThat( testEntity.theInstant ).isEqualTo( DateTimeUtils.roundToSecondPrecision( start, 0 ) );
+		} );
+	}
+
+	@Test
+	@DomainModel(annotatedClasses = TestEntity3.class)
+	@SessionFactory
+	@SkipForDialect( dialectClass = MariaDBDialect.class, reason = "Occasional mismatch in rounding versus our code" )
+	@SkipForDialect( dialectClass = HSQLDialect.class, reason = "Occasional mismatch in rounding versus our code" )
+	@SkipForDialect( dialectClass = DB2Dialect.class, reason = "Occasional mismatch in rounding versus our code" )
+	@SkipForDialect( dialectClass = DerbyDialect.class, reason = "Derby does not support sized timestamp" )
+	@SkipForDialect(dialectClass = SybaseDialect.class, reason = "Because... Sybase...", matchSubTypes = true)
+	void testUsage3(SessionFactoryScope scope) {
+		final Instant start = Instant.now();
+
+		scope.inTransaction( (session) -> {
+			final TestEntity3 testEntity = new TestEntity3();
+			testEntity.id = 1;
+			testEntity.theInstant = start;
+			session.persist( testEntity );
+		} );
+
+		scope.inTransaction( (session) -> {
+			final TestEntity3 testEntity = session.find( TestEntity3.class, 1 );
+			assertThat( testEntity.theInstant ).isEqualTo( DateTimeUtils.roundToSecondPrecision( start, 3 ) );
+		} );
+	}
+
+	@Test
+	@DomainModel(annotatedClasses = TestEntity9.class)
+	@SessionFactory
+	@SkipForDialect( dialectClass = MariaDBDialect.class, reason = "MariaDB only supports precision <= 6" )
+	@SkipForDialect( dialectClass = MySQLDialect.class, reason = "MySQL only supports precision <= 6" )
```

Comment:
```suggestion
	@SkipForDialect( dialectClass = MySQLDialect.class, matchSubTypes = true, reason = "MySQL-like dialects only supports precision <= 6" )
```

Reply:
I'd prefer to leave these as is.  I made that choice explicitly.